"Unknown conversion specified" in test suite with pint 0.18

A new version of pint (0.18) was released last week (2021.10.26), which caused the pyam tests connected to units to fail. 
An easy and quick fix would be to pin the version of `pint <= 0.17`.
Additionally, it might be a good idea to set up nightly tests that run at least once a week so that we catch these dependency changes.

The error messages (e.g. [here](https://github.com/IAMconsortium/units/runs/4153602970?check_suite_focus=true#step:5:106)) look like:
```
iam_units/__init__.py:136: in format_mass
    return format_unit(to_units_container(dict(units), registry=registry), spec)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

unit = <UnitsContainer({'kilogram CO2': 1, 'year': -1})>, spec = ':C'
registry = None, options = {}, fmt = None

    def format_unit(unit, spec, registry=None, **options):
        # registry may be None to allow formatting `UnitsContainer` objects
        # in that case, the spec may not be "Lx"
    
        if not unit:
            if spec.endswith("%"):
                return ""
            else:
                return "dimensionless"
    
        if not spec:
            spec = "D"
    
        fmt = _FORMATTERS.get(spec)
        if fmt is None:
>           raise ValueError(f"Unknown conversion specified: {spec}")
E           ValueError: Unknown conversion specified: :C

/opt/hostedtoolcache/Python/3.10.0/x64/lib/python3.10/site-packages/pint/formatting.py:408: ValueError
```
I also see in places:
```
E           ValueError: Unknown conversion specified: :~
```

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -9,6 +9,18 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        pint-version:
+        # Force specific version of pint
+        - "==0.17"
+        - "==0.18"
+        # No change, i.e. latest
+        - ""
+
+    name: pint${{ matrix.pint-version }}
+
     steps:
     - uses: actions/checkout@v2
 
@@ -21,6 +33,7 @@ jobs:
       run: |
         pip install numpy pytest pytest-cov setuptools-scm
         pip install --editable .
+        pip install pint${{ matrix.pint-version }}
         pytest -rA --verbose --color=yes --cov=iam_units --cov-report=xml
 
     - name: Upload test coverage to Codecov.io

--- a/iam_units/__init__.py
+++ b/iam_units/__init__.py
@@ -111,8 +111,8 @@ def format_mass(obj, info, spec=None):
     info : str
         Any information, e.g. the symbol of a GHG species.
     spec : str, optional
-        Pint formatting specifier such as ':H' (HTML format), ':~' (compact
-        format with symbols), etc.
+        Pint formatting specifier such as "H" (HTML format), "~C" (compact format with
+        symbols), etc.
     """
     spec = spec or obj.default_format
 
@@ -122,15 +122,20 @@ def format_mass(obj, info, spec=None):
     except AttributeError:
         pass  # Already a Unit object
 
-    # Use the symbol for a ':~' spec
+    # Use the symbol if the modifier "~" is in `spec`; else the canonical name. cf.
+    # pint.Unit.__format__()
     method = registry._get_symbol if "~" in spec else lambda k: k
     # Collect the pieces of the unit expression
     units = [[method(key), value] for key, value in obj._units.items()]
 
     # Index of the mass component
     mass_index = list(obj.dimensionality.keys()).index("[mass]")
-    # Append the information to the mass component
+    # Append the information (e.g. species) to the mass component
     units[mass_index][0] += f" {info}"
 
-    # Hand off to pint's formatting
-    return format_unit(to_units_container(dict(units), registry=registry), spec)
+    # Prepare pint.util.UnitsContainer and hand off to pint's formatting. Discard "~"
+    # (used above) and ":" (invalid in pint ≥0.18).
+    return format_unit(
+        to_units_container(dict(units), registry=registry),
+        spec.replace("~", "").lstrip(":"),
+    )

